### PR TITLE
Fix similarity counting issues, with proper test coverage

### PR DESCRIPTION
**Pull request by fwouts**
This fixes the following issues:
- non-JSON payloads always resulted in an `Infinity` score, even if they were identical strings
- the issue also extended to file uploads, which wouldn't match
- headers were not used for comparisons

### CODE REVIEW

File: src/similarity.spec.ts [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from([])
+        },
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(2);
+    expect(
+      computeSimilarity("POST", "/test?a=b", {}, Buffer.from([]), {
```
Comment (kev1n11) [inline src/similarity.spec.ts]: Is it worth testing
- `/test?a=b&c=d` vs. `/test?a=b&c=f` <= 1 same, 1 different
- `/test?a=b&c=d` vs. `/test?a=b&c=d` <= both are the same

File: src/similarity.spec.ts [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        response: DUMMY_RESPONSE
+      })
+    ).toBe(0);
+    expect(
+      computeSimilarity("POST", "/test", {}, Buffer.from("hello world"), {
+        request: {
+          method: "POST",
+          path: "/test",
+          headers: {},
+          body: Buffer.from("hello Kevin")
```
Comment (kev1n11) [inline src/similarity.spec.ts]: Is this a test to see if I read all the test code? 🤔

Comment (kev1n11) [reply src/similarity.spec.ts]: I think Hamming distance might be better for string comparison - no need to do it now tho.

E.g., `hello 1` and `hello 2` should yield a better difference than `hello 1` and `abcdefg`. If I read the code correctly, currently the difference is the same.

File: src/similarity.spec.ts (hunk 1 above)
Comment (fwouts) [reply src/similarity.spec.ts]: Nice catch, this actually caught a bug :)

File: src/similarity.ts [hunk 3]
```diff
@@ -30,36 +31,62 @@ export function computeSimilarity(
...
 
+/**
+ * Returns the number of fields that differ between two objects.
+ */
+function countObjectDifferences(a: object, b: object) {
+  return (diff(a, b) || []).length;
+}
+
+/**
+ * Returns the number of characters that differ between two strings.
```
[the remark this answers is not in the document]
Comment (fwouts) [reply src/similarity.ts]: I clarified the comment. It's a number between 0 and 1, not 0 or 1.

File: src/similarity.spec.ts (hunk 2 above)
Comment (fwouts) [reply src/similarity.spec.ts]: I'll dig into this deeper separately, yeah.

**Pull request merged**